src/main/java/decodes/tsdb/groupedit/TsListSelectPanel.java

```diff
@@ -292,364 +331,462 @@ public Collection<String> getDistinctPart(String part)
+					add = filter == null || filter.equals(timeSeriesIdentifier.getPart(parts[i]));
+					if(!add)
+					{
+						break;
+					}
+				}
+				if(add)
+				{
+					retval.add(timeSeriesIdentifier);
+				}
 			}
+			return retval;
 		}
-		return false;
-	}
-	
-	/**
-	 * Make sure we do not have this combination in the DB already.
-	 * 
-	 * @param siteId
-	 * @param dataTypeId
-	 * @param intervalCode
-	 * @param statisticsCode
-	 * @return true if found a record with the save values, false othewise.
-	 */
-	public boolean verifyConstraints(int siteId, int dataTypeId, 
-							String intervalCode, String statisticsCode)
-	{
-		for(TimeSeriesIdentifier dd : tsidList)
+
+		/**
+		 * Refresh the list with the time series IDs
+		 * from the database.
+		 */
+		private void refresh()
+		{
+			loadTsIds();
+		}
+
+		/**
+		 * Get the time series IDs from the table list
+		 *
+		 * @return ArrayList<TimeSeriesIdentifier>
+		 */
+		public ArrayList<TimeSeriesIdentifier> getTsIdsInList()
+		{
+			ArrayList<TimeSeriesIdentifier> dds = new ArrayList<TimeSeriesIdentifier>();
+			for(TimeSeriesIdentifier vecItem: tsidList)
+				dds.add(vecItem);
+			return dds;
+		}
+
+		/**
+		 * This method is used from the TsGroupDefinition so that
+		 * we set the Time Series group members from the TsGroup obj.
+		 * @param groupList
+		 */
+		public void setTSIDListFromTsGroup(
+				Collection<TimeSeriesIdentifier> ddsIn)
+		{
+			tsidList.clear();
+			tsidList.addAll(ddsIn);
+			sortByColumn(0);
+			fireTableDataChanged();
+		}
+
+
+		/**
+		 * Find out if we have a TSID with the given name and
+		 * param name on the list.
+		 */
+		public boolean ddExistsInList(String siteNameIn, String paramNameIn)
 		{
-			if (dd.getSite().getId().equals(siteId) && 
-				dd.getDataTypeId().equals(dataTypeId) &&
-				dd.getPart("interval").equalsIgnoreCase(intervalCode) &&
-				dd.getPart("statcode").equalsIgnoreCase(statisticsCode)
+			if (siteNameIn != null && paramNameIn != null)
+			{
+				siteNameIn = siteNameIn.trim();
+				paramNameIn = paramNameIn.trim();
+				for(Object vecItem : tsidList)
+				{
+					TimeSeriesIdentifier dd = (TimeSeriesIdentifier) vecItem;
+
+					String siteName = "";
+					if (dd.getSite() != null)
+						siteName = dd.getSite().getDisplayName();
+					String paramName = dd.getPart("param");
+					if (siteName != null)
+						siteName = siteName.trim();
+					if (paramName != null)
+						paramName = paramName.trim();
+					if (siteNameIn.equalsIgnoreCase(siteName) &&
+							paramNameIn.equalsIgnoreCase(paramName))
+						return true;
+				}
+			}
+			return false;
+		}
+
+		/**
+		 * Make sure we do not have this combination in the DB already.
+		 *
+		 * @param siteId
+		 * @param dataTypeId
+		 * @param intervalCode
+		 * @param statisticsCode
+		 * @return true if found a record with the save values, false othewise.
+		 */
+		public boolean verifyConstraints(int siteId, int dataTypeId,
+										 String intervalCode, String statisticsCode)
```

Comment:
Quite a bit of this looks to be CWMS specific but doesn't actually have any usages.